test-suite/src/test/scala/org/scalajs/testsuite/javalib/ListTest.scala

```diff
@@ -0,0 +1,270 @@
+      al.add("three")
+
+      al.set(1, "four")
+      expect(al.get(0)).toEqual("one")
+      expect(al.get(1)).toEqual("four")
+      expect(al.get(2)).toEqual("three")
+
+      expectIndexOutOfBoundsException(al.set(-1, ""))
+      expectIndexOutOfBoundsException(al.set(al.size, ""))
+    }
+
+    it("should give proper iterator over elements") {
+      val al = listFactory.empty[String]
+      al.add("one")
+      al.add("two")
+      al.add("three")
+
+      val elements = al.iterator
+      expect(elements.hasNext).toBeTruthy
+      expect(elements.next).toEqual("one")
+      expect(elements.hasNext).toBeTruthy
+      expect(elements.next).toEqual("two")
+      expect(elements.hasNext).toBeTruthy
+      expect(elements.next).toEqual("three")
+      expect(elements.hasNext).toBeFalsy
+    }
+
+    it("should give proper list iterator over elements") {
+      val al = listFactory.empty[String]
+      al.add("one")
+      al.add("two")
+      al.add("three")
+
+      val elements = al.listIterator
+      expect(elements.hasPrevious).toBeFalsy
+      expect(elements.hasNext).toBeTruthy
+      expect(elements.next).toEqual("one")
+      expect(elements.hasPrevious).toBeTruthy
+      expect(elements.hasNext).toBeTruthy
+      expect(elements.next).toEqual("two")
+      expect(elements.hasPrevious).toBeTruthy
+      expect(elements.hasNext).toBeTruthy
+      expect(elements.next).toEqual("three")
+      expect(elements.hasPrevious).toBeTruthy
+      expect(elements.hasNext).toBeFalsy
+      expect(elements.previous).toEqual("three")
+      expect(elements.previous).toEqual("two")
+      expect(elements.previous).toEqual("one")
+    }
+
+    it("should add elements at a given index") {
+      val al = listFactory.empty[String]
+      al.add("one") // ("one")
+      al.add(0, "two") // ("two", "one")
+      al.add(1, "three") // ("two", "three", "one")
+
+      expect(al.get(0)).toEqual("two")
+      expect(al.get(1)).toEqual("three")
+      expect(al.get(2)).toEqual("one")
+
+      expectIndexOutOfBoundsException(al.add(-1, ""))
+      expectIndexOutOfBoundsException(al.add(al.size + 1, ""))
+    }
+
+    it("should give the first index of an element") {
+      val al = listFactory.empty[String]
+      al.add("one")
+      al.add("two")
+      al.add("three")
+      al.add("one")
+      al.add("two")
+      al.add("three")
+
+      expect(al.indexOf("one")).toEqual(0)
+      expect(al.indexOf("two")).toEqual(1)
+      expect(al.indexOf("three")).toEqual(2)
+      expect(al.indexOf("four")).toEqual(-1)
+    }
+
+    it("should give the last index of an element") {
+      val al = listFactory.empty[String]
+      al.add("one")
+      al.add("two")
+      al.add("three")
+      al.add("one")
+      al.add("two")
+      al.add("three")
+
+      expect(al.lastIndexOf("one")).toEqual(3)
+      expect(al.lastIndexOf("two")).toEqual(4)
+      expect(al.lastIndexOf("three")).toEqual(5)
+      expect(al.indexOf("four")).toEqual(-1)
+    }
+
+
+  }
```

Comment:
There should be tests for `subList`, including testing propagations of changes from the parent list to the sub list and vice versa.